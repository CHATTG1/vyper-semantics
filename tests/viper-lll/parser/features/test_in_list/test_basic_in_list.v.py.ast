%pgm(, ,
  %fdecl(%@public, testin, %param(x, %num), %bool,
    %assign(%var(y), 1)
    %assign(%var(s), %list(1 2 3 4))
    %if(%compareop(%in, %binop(+, %var(x), 1), %var(s)),
      %return(true))
    %return(false))
)
