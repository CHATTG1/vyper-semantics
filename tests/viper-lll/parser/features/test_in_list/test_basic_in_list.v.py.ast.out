$%seq ( $return ( 0 , $lll ( $%seq ( $%seq ( $mstore ( 28 , $calldataload ( 0 ) ) , $mstore ( 32 , 1461501637330902918203684832716283019655932542976 ) , $mstore ( 64 , 170141183460469231731687303715884105727 ) , $mstore ( 96 , -170141183460469231731687303715884105728 ) , $mstore ( 128 , 1701411834604692317316873037158841057270000000000 ) , $mstore ( 160 , -1701411834604692317316873037158841057280000000000 ) ) , $if ( $eq ( $mload ( 0 ) , 1361642264 ) , $%seq ( $calldatacopy ( 320 , 4 , 32 ) , $assert ( $iszero ( $callvalue ) ) , $clamp ( $mload ( 96 ) , $calldataload ( $add ( 4 , 0 ) ) , $mload ( 64 ) ) , $mstore ( 352 , 1 ) , $with ( $_L , 384 , $%seq ( $mstore ( $add ( $mul ( 32 , $uclamplt ( 0 , 4 ) ) , $_L ) , 1 ) , $mstore ( $add ( $mul ( 32 , $uclamplt ( 1 , 4 ) ) , $_L ) , 2 ) , $mstore ( $add ( $mul ( 32 , $uclamplt ( 2 , 4 ) ) , $_L ) , 3 ) , $mstore ( $add ( $mul ( 32 , $uclamplt ( 3 , 4 ) ) , $_L ) , 4 ) ) ) , $if ( $eq ( 1 , $%seq ( $mstore ( 512 , 0 ) , $with ( _result , 512 , $repeat ( 256 , 0 , 4 , $if ( $eq ( $clamp ( $mload ( 96 ) , $add ( $mload ( 320 ) , 1 ) , $mload ( 64 ) ) , $mload ( $add ( 384 , $mul ( 32 , $mload ( 256 ) ) ) ) ) , $%seq ( $mstore ( _result , 1 ) , $break ) ) ) ) , $mload ( 512 ) ) ) , $%seq ( $%seq ( $mstore ( 0 , 1 ) , $return ( 0 , 32 ) ) ) ) , $%seq ( $mstore ( 0 , 0 ) , $return ( 0 , 32 ) ) , $stop ) ) ) , 0 ) ) )
